utils/maptestmessages.pl

```diff
@@ -1,28 +1,36 @@
 #!/usr/bin/env perl
 
-use 5.14.2;
+use 5.16.0;
 use warnings;
-use Zonemaster;
+use Zonemaster::Engine;
 
 # page header
 print "# Mapping test messages to test module\n\n";
 
 # table header
-print "| Log message identifier        | Implemented test case         |\n";
-print "|:------------------------------|:------------------------------|\n";
+print "| Log message identifier                         | Implemented test case          |\n";
+print "|:-----------------------------------------------|:-------------------------------|\n";
 
 # table content
-foreach my $module ( 'Basic', sort { $a cmp $b } Zonemaster::Test->modules ) {
-    my $full = "Zonemaster::Test::$module";
+my %links;
+foreach my $module ( 'Basic', sort { fc $a cmp fc $b } Zonemaster::Engine::Test->modules ) {
+    my $full = "Zonemaster::Engine::Test::$module";
     my $ref  = $full->metadata;
-    while ( my ($key, $list) = each %$ref ) {
+    for my $key (sort keys %$ref) {
+        my $list = $ref->{$key};
         for my $tag (map { uc( $module ) . ':' . $_ } sort { $a cmp $b } @$list) {
-	    # printf "%46s -> %s::%s\n", $tag, $module, $key;
-	    my $testmodule = sprintf("%s::%s", $module, $key);
-	    my $testlevel = "$module-TP";
-	    my $testcase = "$key.md";
-	    my $testlink = "$testlevel/$testcase";
-	    printf "| %-30s | [%s](%s) |\n", $tag, $testmodule, $testlink;
```

Comment:
Why would you want to remove the link to the test case specification?

Reply:
The link target is just relocated to the page footer. That saves some horizontal space in text mode.

Reply:
But is text mode realistically the proper use case for this file? The whole point is that you can search for the log message, and find the exact specification of why you see the message. This adds another indirection of doing this.

Reply:
I didn't think of that use case. I figured people would enter the log message into Google and end up on Github: https://google.com/search?q=zonemaster+BASIC%3ADOMAIN_NAME_TOO_LONG